pkg/fftypes/tokenpool.go

```diff
@@ -0,0 +1,33 @@
+// Copyright © 2021 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fftypes
+
+type TokenType = LowerCasedType
+
+const (
+	TokenTypeFungible    TokenType = "fungible"
+	TokenTypeNonFungible TokenType = "nonfungible"
+)
+
+type TokenPool struct {
+	ID         *UUID     `json:"id,omitempty"`
+	Type       TokenType `json:"type"`
+	Namespace  string    `json:"namespace,omitempty"`
+	Name       string    `json:"name,omitempty"`
+	ProtocolID string    `json:"protocol_id,omitempty"`
```

Comment:
Important that we're consistent on our API
```suggestion
	ProtocolID string    `json:"protocold,omitempty"`
```

Note in `transactions.go` I see `protocolID` with a capital `D` 🤔  ... we need to sort out whether we're an `somethingID` or `somethingId` in this PR, and stick to it from now on.

Reply:
Have confirmed `streamID`, `backendID`, `protocolID`, `requestID` all are consistent... so sounds like we have a pattern. So it's now or never to change it to `Id` - otherwise we have our horse.

Reply:
I think we have some sort of Go style plugin that actually enforces `ID` over `Id`